### PR TITLE
Feat/340 public profile page

### PR DESCRIPTION
**Pull request by Clemensito**
# Summary
Created a public profile view, depending if you are friends or not you can see the users posts and follower count (which has to be updated in a future PR) as well as a follow button (Which is also out of scope for this PR). Implemented privacy for the user i.e if the user is not a friend you cannot see their posts

# What

Added ViewProfile screen and viewModel. The VM gets the info from user collection and outfit collection to be shown in the screen. As said the follow button is not functionnal yet and will have to be implemented later on. 
Also added the profile picture of the user upon search since I thought it would be a small improvement

- UI
  - Added a public profile screen containing username, profile picture, friend count and follow button (both not functionnal) and posts (only if mutuals, and are clickable)
  - Profile picture upon search
- ViewModel
  - As described above, ViewModel retrieving pretty much all info needed, as well as checking the requesters friend list to see if him and the searched user are friends. If they are the posts are shown, else a text stating they need to add the user to see their posts
- Documentation
  - Kdoc comments added for the new class and methods
- Tests
  - Unit tests added for ViewUser VM and screen, updated for Search screen and profile
  - Integration tests same as above

# Why
Needed for UX 

# Screenshots / Videos

image

image

image

# Checklist
- [x] Tests: added or updated (unit/instrumented/automation) as appropriate
- [x] Documentation: README/usage/docs updated where relevant
- [x] Manual verification: app run and basic flows checked (list steps in "What")

### CODE REVIEW

File: app/src/main/java/com/android/ootd/MainActivity.kt [hunk 1]
```diff
@@ -336,6 +341,21 @@ fun OOTDApp(
...
+                      if (userId != null) {
+                        ViewUserProfile(
+                            userId = userId,
+                            onBackButton = { navigationActions.goBack() },
+                            onPostClick = { postId ->
+                              navigationActions.navigateTo(Screen.PostView(postId))
+                            })
+                      }
+                    }
+
```
Comment (j-meric) [inline app/src/main/java/com/android/ootd/MainActivity.kt]: **Praise:** Good job implementing the navigation in mainactivity.

File: app/src/main/java/com/android/ootd/ui/account/ViewUserViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      refresh(friendId)
+    }
+  }
+
+  /**
+   * Refreshes the user profile data from the repositories.
+   *
+   * @param friendId The unique identifier of the user whose profile is being viewed
+   */
+  private fun refresh(friendId: String) {
```
Comment (j-meric) [inline app/src/main/java/com/android/ootd/ui/account/ViewUserViewModel.kt]: **Praise** Great job thinking of this very vital but easily forgettable aspect of refreshing the screen for the viewUserViewModel

File: app/src/androidTest/java/com/android/ootd/ui/account/ViewUserScreenTest.kt [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+    composeTestRule.onNodeWithTag(ViewUserScreenTags.POSTS_SECTION_TAG).assertIsDisplayed()
+    composeTestRule
+        .onNodeWithTag(ViewUserScreenTags.POSTS_SECTION_TAG)
+        .assertTextContains("Posts :")
+    composeTestRule.onAllNodesWithTag(ViewUserScreenTags.POST_TAG).assertCountEquals(3)
+  }
+
+  @Test
+  fun screen_displays_add_friend_message_when_not_friend() {
```
Comment (j-meric) [inline app/src/androidTest/java/com/android/ootd/ui/account/ViewUserScreenTest.kt]: **Nitpick**: Same as above for when not a friend

File: app/src/test/java/com/android/ootd/ui/account/ViewUserViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+    viewModel.update("friend2")
+    advanceUntilIdle()
+
+    state = viewModel.uiState.value
+    assertEquals("user2", state.username)
+    assertEquals("", state.profilePicture)
+    assertFalse(state.isFriend)
+    assertEquals(0, state.friendPosts.size)
+  }
```
Comment (j-meric) [inline app/src/test/java/com/android/ootd/ui/account/ViewUserViewModelTest.kt]: **Praise**: Overall comprehensive tests that cover most edge cases, well done.

File: app/src/main/java/com/android/ootd/ui/account/ViewUserViewModel.kt (hunk 2 above)
Comment (Clemensito) [reply app/src/main/java/com/android/ootd/ui/account/ViewUserViewModel.kt]: Thanks !

**Review by j-meric — APPROVED** [review]
Great job, the code is now more modular and as a result you saved a few hundred lines on the pr! There is one last sonar issue, about a commented out line, which is not good practice to merge with main, but other than that your code is good to go.
